cloudferrylib/os/actions/verify_vms.py

```diff
@@ -0,0 +1,40 @@
+# Copyright (c) 2015 Mirantis Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the License);
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an AS IS BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+# implied.
+# See the License for the specific language governing permissions and#
+# limitations under the License.
+
+
+from cloudferrylib import utils
+from cloudferrylib.base import exception
+from cloudferrylib.base.action import action
+
+
+class VerifyVms(action.Action):
+    def run(self, **kwargs):
+        if not kwargs.get('info'):
+            raise exception.AbortMigrationError(
+                "No information from destination cloud."
+                "Something has been broken on early steps."
+            )
+        dst_info = kwargs['info']
+        search_opts = {'search_opts': kwargs.get('search_opts', {})}
+        search_opts.update(kwargs.get('search_opts_tenant', {}))
+        compute_resource = self.cloud.resources[utils.COMPUTE_RESOURCE]
+        src_info = compute_resource.read_info(**search_opts)
+        old_ids = set(dst_inst['meta']['old_id']
+                      for dst_inst in dst_info['instances'].values())
+        for src_inst_id in src_info['instances']:
+            if src_inst_id not in old_ids:
+                raise exception.AbortMigrationError(
```

Comment:
I think would be better to list VMs which failed to migrate, not the first one. Maybe incorrectly migrated VMs can be restored/updated manually if there are just few.
Also, simply comparing instance IDs is obviously not enough for such a verification.